### PR TITLE
org mem indexing

### PR DESCRIPTION
**Pull request by yingjieMiao**
@leogrim 

next steps: augment `KeepIndexable` and `LibraryIndexable` with `orgId`. So that we can update `VisibilityEvaluator`. 

Those are largely orthogonal to this PR, I think.

### CODE REVIEW

File: kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.keepit.search.index.{ Searcher, Indexable, FieldDecoder }
+import com.keepit.search.util.LongArraySet
+import org.apache.lucene.index.Term
+
+object OrganizationMembershipFields {
+  val userField = "u"
+  val userIdField = "uid"
+  val orgField = "o"
+  val orgIdField = "oid"
+
```
Comment (leogrim) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala:14]: Perhaps we should at least index the user's role? (I mean a value field)

**Comment by leogrim** [comment]
Awesome! We probably want convenience methods defined on `OrganizationMembershipIndexable`, similar to those defined on `LibraryMembershipIndexable` :)

File: kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala:14]: from visibility point of view, does role make a difference? For current usage, the field will not be used, right?

**Comment by yingjieMiao** [comment]
convenience methods will be added as they are needed.

File: kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  val decoders: Map[String, FieldDecoder] = Map.empty
+}
+
+object OrganizationMembershipIndexable {
+  import OrganizationMembershipFields._
+
+  def getOrgsByMember(orgMemSearcher: Searcher, memberId: Id[User]): Set[Long] = {
+    LongArraySet.from(orgMemSearcher.findSecondaryIds(new Term(userField, memberId.id.toString), orgIdField).toArray())
+  }
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/index/graph/organization/OrganizationMembershipIndexable.scala:23]: one convenient method so far ...

**Pull request merged**